firefox-ios/Client/Frontend/Microsurvey/Prompt/MicrosurveyPromptState.swift

```diff
@@ -31,9 +31,12 @@ struct MicrosurveyPromptState: StateType, Equatable {
     }
 
     static let reducer: Reducer<Self> = { state, action in
-        // TODO: FXIOS-9068 Need to test this experience with multiwindow
+        guard action.windowUUID == .unavailable || action.windowUUID == state.windowUUID else { return state }
+
         switch action.actionType {
-        case MicrosurveyPromptMiddlewareActionType.initialize(let model):
+        case MicrosurveyPromptMiddlewareActionType.initialize:
+//            let model = (action as? MicrosurveyPromptMiddlewareAction)?.microsurveyModel
+            let model = action.payload as? MicrosurveyModel
```

Comment:
@OrlaM  For demoing purposes, I added another commit to discuss using payload instead of our current implementation. The commented out code is to demonstrate what we are doing today. Currently, a lot of actions tend to use a singular built-in object types that we can pull from the payload (like we show here with `MicrosurveyModel`.

I was curious in whether we also discussed a simple payload with `Any?` type. The downside I see is that we may need to check the type as an extra step, but already do a lot of checks with the action types as well. For complex payloads, we can create a separate struct to associate with that specific action. 
```
struct CloseTabPayload {
   let panelType: panelType,
   let tabUUID: tabUUID
} 
let payload = CloseTabPayload(panelType: panelType,
                                            tabUUID: tabUUID)
let action = TabPanelViewAction(windowUUID: windowUUID,
                                            actionType: TabPanelViewActionType.closeTab, 
                                            payload: payload)
```
Let me know what you think!